mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/ConfigLookUp.java

```diff
@@ -0,0 +1,43 @@
+package org.arquillian.smart.testing.mvn.ext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.arquillian.smart.testing.configuration.ConfigurationLoader.SMART_TESTING_YAML;
+import static org.arquillian.smart.testing.configuration.ConfigurationLoader.SMART_TESTING_YML;
+
+class ConfigLookUp {
+
+    private final File executionRootDir;
+
+    ConfigLookUp(String executionRootDir) {
+        this.executionRootDir = new File(executionRootDir);
+    }
+
+    File getFirstDirWithConfigOrProjectRootDir() {
+        return getFirstDirWithConfigOrProjectRootDir(executionRootDir, new File(System.getenv("MAVEN_PROJECTBASEDIR")));
+    }
+
+    private File getFirstDirWithConfigOrProjectRootDir(File projectDir, File multiModuleProjectDirectory) {
+        if (isSameFile(projectDir.toPath(), multiModuleProjectDirectory.toPath())) {
+            return projectDir;
+        }
+
+        if (projectDir.isDirectory() && new File(projectDir, "pom.xml").exists() &&
```

Comment:
General thought here:

- should this really belong to `mvn` part? I think configuration as such is a general concept.
- this implementation couples us with Maven - maybe we should split it up then. Not for now, but can result in the follow-up story

WDYT @MatousJobanek?

Reply:
I was thinking about the same during our discussion today. 
IIRC, the reason why it was done it in this way is that ST is Maven extension, so it kind of reflects Maven behavior/project structure.
But, you are right about it that we probably couple with Maven too much.